logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -194,6 +199,33 @@ class MLSConversationDataSource(
         }
     }
 
+    override suspend fun requestToJoinGroupByExternalCommit(groupID: GroupID, conversationId: QualifiedID): Either<CoreFailure, Unit> {
+        kaliumLogger.d("Requesting to re-join MLS group $groupID via external commit")
+        return wrapApiRequest {
+            conversationApi.fetchGroupInfo(idMapper.toApiModel(conversationId))
+        }.flatMap {
+            mlsClientProvider.getMLSClient().flatMap { mlsClient ->
+                wrapMLSRequest {
+                    mlsClient.joinByExternalCommit(it)
+                }.flatMap { commitBundle ->
+                    sendCommitBundle(groupID, commitBundle)
+                }.flatMap {
+                    wrapMLSRequest {
+                        mlsClient.mergePendingGroupFromExternalCommit(idMapper.toCryptoModel(groupID))
+                    }
+                }
```

Comment:
When testing this I noticed that we also attempt to clear the commit when fetching the group info fails, which will trigger a core crypto exception.

So I suggest moving in here like this:

```suggestion
                }.fold({
                     wrapMLSRequest {
                         mlsClient.clearPendingGroupExternalCommit(idMapper.toCryptoModel(groupID))
                     }
                }, {
                    wrapMLSRequest {
                        mlsClient.mergePendingGroupFromExternalCommit(idMapper.toCryptoModel(groupID))
                    }
                })
```